waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/SessionFactorySupplier.java

```diff
@@ -31,15 +34,40 @@
 public class SessionFactorySupplier implements Supplier<SessionFactory> {
   private static final Logger LOG = LoggerFactory.getLogger(SessionFactorySupplier.class);
 
+  @VisibleForTesting
+  static class CustomSessionFactory extends DefaultSessionFactory {
+    final int sshTimeout;
+
+    private CustomSessionFactory(int sshTimeout) {
+      this.sshTimeout = sshTimeout;
+    }
+
+    @Override
+    public Session newSession() throws JSchException {
+      Session session = super.newSession();
+      session.setTimeout(sshTimeout);
+      return session;
+    }
+  }
+
   private final int sshPort;
   private final String knownHosts;
   private final List<String> identityKeys;
+  private final int sshTimeout;
   private DefaultSessionFactory sessionFactory = null;
 
+  @Deprecated
   public SessionFactorySupplier(int sshPort, String knownHosts, List<String> identityKeys) {
+    this(sshPort, knownHosts, identityKeys, 0);
+  }
+
+  public SessionFactorySupplier(int sshPort, String knownHosts, List<String> identityKeys, int sshTimeout) {
+    Preconditions.checkArgument(0 <= sshPort && sshPort <= 65535, "Invalid SSH port number " + sshPort);
```

Comment:
Why do we have this check both here and in the TunnelHandler  but we only have the timeout check here and not in the TunnelHandler ?

Reply:
I'm improving the code, the old version does not check... the constructor is public

Reply:
OK